File: internal/miner/testdata/stringly_drift/typo_case.go
// Package x demonstrates a stringly-typed closed-enum drift defect.
// Status is a named string type with a closed const set.
// The switch has a typo case literal "activ" that matches no const value.
package x

// Status is a named string type — a closed enum.
type Status string

const (
	StatusActive   Status = "active"
	StatusInactive Status = "inactive"
	StatusPending  Status = "pending"
)

// handleStatus dispatches on Status.
// BUG: case "activ" is a typo; no const value of Status equals "activ".
func handleStatus(status Status) string {
	switch status {
	case "activ": // typo — should be "active" (StatusActive)
		return "do-active-thing"
	case "inactive":
		return "do-inactive-thing"
	case "pending":
		return "do-pending-thing"
	default:
		return "unhandled"
	}
}

File: internal/miner/testdata/stringly_ts_drift/typo_case.ts
// Positive fixture: 'activ' is a typo — should be 'active'.
// Expected: 1 type-A lead at the case 'activ' line.

type Status = 'active' | 'inactive' | 'pending';

function handleStatus(s: Status): string {
  switch (s) {
    case 'activ':       // typo — not a member of Status
      return 'typo branch';
    case 'inactive':
      return 'inactive branch';
    case 'pending':
      return 'pending branch';
    default:
      return 'unknown';
  }
}
